### DISCUSSION TITLE
Enhancement: Manually set the history message count and length

### DISCUSSION THREAD

**Discussion by CXwudi** [danny-avila/LibreChat#2000] [Feature Requests & Suggestions]
### What features would you like to see added?

Is there a way to manually set the history message count and set the history message length threshold?

And even better set both threshold to unlimit so that all history messages will be carried forward in a chat, regardless of the model in use

### More details

Please see the screenshots below

### Which components are impacted by your request?

UI, Endpoints, Plugins

### Pictures

I am actually looking for something like this, these screenshots are taken from [lobe-chat](https://github.com/lobehub/lobe-chat)

![image](https://github.com/danny-avila/LibreChat/assets/17377423/5761de91-ec48-4b96-a497-ff89483faba7)
![image](https://github.com/danny-avila/LibreChat/assets/17377423/4a1ebbbe-18ec-4e16-b632-ba08cb3d3d78)

Here are another screenshots taken from [ChatGPT-Next-Web](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web)

![image](https://github.com/danny-avila/LibreChat/assets/17377423/b318a687-29b2-4b02-a924-22b6c5606e35)
![image](https://github.com/danny-avila/LibreChat/assets/17377423/d3db6993-a18f-4ebf-989a-01c275528d7a)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hijos** [comment]
I also think this feature is necessary.

我也觉得需要这个功能。

**Reply by CXwudi** [reply]
Just use LobeChat, I already switched to it

**Reply by hijos** [reply]
> Just use LobeChat, I already switched to it

haha, thank you.But I love the forking messages and conversations feature of librechat.

**Reply by danny-avila** [reply]
you can already set a context limit: 

![image](https://github.com/user-attachments/assets/e1b5e687-d0d1-45bd-a761-b5c2b2046137)

Setting a limit by number of messages would be simple enough, too, would work similarly
